### PR TITLE
va-button: Add USWDS v3 variation

### PR DESCRIPTION
**Pull request by jamigibbs**
## Chromatic

https://1515-va-button-v3--60f9b557105290003b387cd5.chromatic.com/?path=/docs/uswds-va-button--primary

## Description
This PR adds the USWDS v3 button variation to `va-button`.

Sketch: https://www.sketch.com/s/610156b6-f281-4497-81f3-64454fc72156/a/P1Dzynv
Closes https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1515

## Testing done
Storybook

## Screenshots

Screenshot 2023-03-10 at 10 29 12 AM

Screenshot 2023-03-10 at 10 29 42 AM

Screenshot 2023-03-10 at 10 30 18 AM

Screenshot 2023-03-10 at 10 30 31 AM

Screenshot 2023-03-10 at 10 30 44 AM

Screenshot 2023-03-10 at 10 30 58 AM

Screenshot 2023-03-10 at 10 31 30 AM

## Acceptance criteria
- [ ] The USWDS v3 variation has been added to `va-button`

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

File: packages/web-components/src/components/va-button/va-button.scss [hunk 1]
```diff
@@ -1,10 +1,25 @@
...
+
+@import '../../global/formation_overrides';
+
+:host([disabled]:not([disabled='false'])) button {
+  pointer-events: none;
+}
+
+:host([disabled]:not([disabled='false'])) {
+  cursor: not-allowed;
+}
```
Comment (jamigibbs) [inline packages/web-components/src/components/va-button/va-button.scss:13]: The [USWDS version](https://designsystem.digital.gov/components/button/) did not have any cursor or pointer events disabled for the `disabled` button which I thought was a little strange so I am adding it here to work both for v1 and v3.

File: packages/web-components/src/components/va-button/va-button.tsx [hunk 2]
```diff
@@ -121,29 +132,62 @@ export class VaButton {
...
-      </Host>
-    );
+    if (uswds) {
+      const buttonClass = classnames({
+        'usa-button': true,
+        'usa-button--secondary': secondary,
+        'usa-button--big': big,
+        'usa-button--outline': back || outline,
+      });
+
```
Comment (jamigibbs) [inline packages/web-components/src/components/va-button/va-button.tsx:152]: There are three "color variations" that we are not incorporating at this time:

- usa-button--accent-cool
- usa-button--accent-warm
- usa-button--base

Screenshot 2023-03-10 at 10 42 56 AM

When or if we do decide to add them, my recommended approach would be to add a new prop `variation` that would accept one of the three values `accent-cool`, `accent-warm`, or `base`.

And that value can then be used to merge into the `buttonClass` string. For example (with possibly some additional conditional work based on other prop settings):

```
buttonClass += ` usa-button--${variation}`
```

File: packages/web-components/src/mixins/focusable.css [hunk 3]
```diff
@@ -1,7 +1,7 @@
 /* From the USWDS styles: */
 /*   https://github.com/uswds/uswds/blob/3dc296ec56cd621fe52d918701fd94621d96a198/src/stylesheets/core/mixins/_focus.scss#L12-L13 */
 button:not([disabled]):focus,
-button:not([disabled]):active,
+:host(:not([uswds])) button:not([disabled]):active,
```
Comment (jamigibbs) [inline packages/web-components/src/mixins/focusable.css:4]: The USWDS version does not have the focus halo for active states so I have removed it from the uswds variation but I'm confirming if that's what we want too. If we want the active state to have the gold focus, I'll revert this.

File: packages/web-components/src/components/va-button/va-button.tsx (hunk 2 above)
Comment (jhonnyoddball) [reply packages/web-components/src/components/va-button/va-button.tsx:152]: I like the idea

**Comment by jamigibbs** [comment]
@GnatalieH When you have a chance, do you mind taking a look at this? I also wasn't sure if there was a Sketch link yet for v3 buttons.

File: packages/web-components/src/mixins/focusable.css (hunk 3 above)
Comment (GnatalieH) [reply packages/web-components/src/mixins/focusable.css:4]: @jamigibbs I think we want to mimic the style of the USWDS button as much as possible, so it's OK to remove the focus halo from the active states.

**Comment by GnatalieH** [comment]
@jamigibbs [here is an updated link](https://www.sketch.com/s/610156b6-f281-4497-81f3-64454fc72156/a/P1Dzynv) to the Sketch file, just FYI.

**Review by GnatalieH — APPROVED** [review]
@jamigibbs I included a Big Secondary button in my mockups, too. Otherwise this looks good. But I'm not sure the Back and Continue and Disabled buttons should be included. @humancompanion-usds can you comment on that? The Disabled button probably is needed for loading states.

**Comment by caw310** [comment]
Discussed and decided during stand up that we will keep the continue and back buttons.

**Pull request merged**